meta.py

```diff
@@ -682,31 +675,48 @@ def set_result(msg_short, msg_long):
     set_result('Success', '')
 
 
-@rule.make()
-def rule_copy_user_metadata(ctx, source, target):
-    copy_user_metadata(ctx, source, target)
-
-
 def copy_user_metadata(ctx, source, target):
     """
-    Copy the user metadata of a collection to another collection.
+    Copy the user metadata (AVUs) of a collection to another collection.
+
+    This only copies user metadata, so it ignores system metadata.
 
     :param ctx:    Combined type of a callback and rei struct
     :param source: Path of source collection.
     :param target: Path of target collection.
     """
     try:
-        # Retrieve all user metadata on source collection.
-        iter = genquery.row_iterator(
-            "META_COLL_ATTR_NAME, META_COLL_ATTR_VALUE",
-            "COLL_NAME = '{}' AND META_COLL_ATTR_NAME = '{}%'".format(source, constants.UUUSERMETADATAPREFIX),
-            genquery.AS_LIST, ctx
-        )
-
-        # Set user metadata on target collection.
-        for row in iter:
-            avu.associate_to_coll(ctx, target, row[0], row[1])
-
-        log.write(ctx, "rule_copy_user_metadata: copied user metadata from <{}> to <{}>".format(source, target))
+        # Retrieve all AVUs inside source collection.
+        user_metadata = list(avu.inside_coll(ctx, source, recursive=True))
+
+        # Group AVUs by entity and filter system metadata.
+        grouped_user_metadata = {}
+        for path, type, attribute, value, unit in user_metadata:
+            if not attribute.startswith(constants.UUORGMETADATAPREFIX) and unit != constants.UUFLATINDEX and not unit.startswith(constants.UUUSERMETADATAROOT + '_'):
+                grouped_user_metadata.setdefault(path, {"type": type, "avus": []})
+                grouped_user_metadata[path]["avus"].append((attribute, value, unit))
+
+        # Generate metadata operations.
+        for path, item in grouped_user_metadata.items():
+            operations = {
+                "entity_name": path.replace(source, target + "/original"),
```

Comment:
A bit far-fetched, but could there be a subfolder `path` with a second match of `source` so then the part would be replaced in `path` twice? For example:
```python3
source = '/tempZone/home/research-initial'
path = '/tempZone/home/research-initial/tempZone/home/research-initial/some-folder'
```

Reply:
Good catch!